test/bn256_test.cpp

```diff
@@ -1,179 +1,105 @@
-            std::cout << "unmarshal_success" << std::endl;
-            break;
-        case bn256::unmarshal_malformed_point:
-            std::cout << "unmarshal_malformed_point" << std::endl;
-            break;
-        case bn256::unmarshal_coordinate_exceeds_modulus:
-            std::cout << "unmarshal_coordinate_exceeds_modulus" << std::endl;
-            break;
-    }
-}
-
-int test_g1_marshall() {
-    std::cout << std::endl << "test_g1_marshall:" << std::endl;
-
-    int512_t k;
-    bn256::g1 ga{}, gb{};
+TEST_CASE("test g1 marshall", "[bn256]"){
+    auto [_, ga] = bn256::ramdom_g1();
+    std::array<uint8_t, 64> ma, mb;
+    ga.marshal(ma);
 
-    std::tie(k, ga) = bn256::ramdom_g1();
-    auto ma = ga.marshal();
+    bn256::g1 gb{};
 
-    auto status = gb.unmarshal(ma);
-    if (status != bn256::unmarshal_success){
-        print_umarshall_status(status);
-        return test_fail;
-    }
-
-    auto mb = gb.marshal();
-    if (ma != mb) {
-        std::cout << "bytes are different" << std::endl;
-        return test_fail;
-    }
+    REQUIRE( gb.unmarshal(ma) == std::error_code{});
 
-    return test_pass;
+    gb.marshal(mb);
+    CHECK(ma == mb);
 }
 
-int test_g2_marshall() {
-    std::cout << std::endl << "test_g2_marshall:" << std::endl;
+TEST_CASE("test g2 marshall", "[bn256]"){
+    auto [_, ga] = bn256::ramdom_g2();
+    std::array<uint8_t, 128> ma, mb;
+    ga.marshal(ma);
 
-    int512_t k;
-    bn256::g2 ga{}, gb{};
-
-    std::tie(k, ga) = bn256::ramdom_g2();
-
-    auto ma = ga.marshal();
-    auto status = gb.unmarshal(ma);
-    if (status != bn256::unmarshal_success){
-        print_umarshall_status(status);
-        return test_fail;
-    }
-    auto mb = gb.marshal();
-
-    if (ma != mb) {
-        std::cout << "bytes are different" << std::endl;
-        return test_fail;
-    }
-    return test_pass;
+    bn256::g2 gb{};
+    REQUIRE( gb.unmarshal(ma) ==  std::error_code{});
+    gb.marshal(mb);
+    CHECK(ma == mb);
 }
 
-int test_bilinearity() {
-    std::cout << std::endl << "test_bilinearity:" << std::endl;
-
-    int512_t a, b;
-    bn256::g1 p1{}, c1{bn256::curve_gen};
-    bn256::g2 p2{}, c2{bn256::twist_gen};
+TEST_CASE("test bilinearity", "[bn256]"){
+    bn256::g1 c1{bn256::curve_gen};
+    bn256::g2 c2{bn256::twist_gen};
 
-    for (auto i = 0; i < 2; ++i) {
-        std::tie(a, p1) = bn256::ramdom_g1();
-        std::tie(b, p2) = bn256::ramdom_g2();
+    for (auto i = 0U; i < 2; ++i) {
+        auto [a, p1] = bn256::ramdom_g1();
+        auto [b, p2] = bn256::ramdom_g2();
         bn256::gt e1 = bn256::pair(p1, p2);
-        bn256::gt e2 = bn256::pair(c1, c2);
-        e2.scalar_mult(e2, a);
-        e2.scalar_mult(e2, b);
 
-        if (e1 != e2) {
-            std::cout << "bad pairing result" << std::endl;
-            return test_fail;
-        }
+        bn256::gt e2 = bn256::pair(c1, c2);
+        e2 = e2.scalar_mult(a);
+        e2 = e2.scalar_mult(b);
+        CHECK(e1 == e2);
```

Comment:
It might be useful put the original test failure message as comment for at least complicated tests to help understand the tests.

Reply:
Resolved with https://github.com/AntelopeIO/bn256/pull/4/commits/ec6fc95c777a49b468aee12804c7bc4d0972d613